### ISSUE TITLE
Add continuous color scale to strip, box, and violin plots

### ISSUE THREAD

**Issue by joelostblom** [plotly/plotly.py#1938]
Currently, strip/box/violin plots only work with categorical/discrete colormaps. This is sufficient for the violins and boxes but for the points that can be included in all three plots, it would be useful to color by a continuous feature and to add a colorbar to the plot. I looked around a bit and it seems like the go.Box constructor needs to have the coloraxis attrbute added to it (and I suspect something would need to be added to avoid coloring the boxes/violins in the same colors as the points, since they are colored the same for categorical variables).

The example below illustrates the issue by trying to color by the sepal_width variable:

```python
import plotly.express as px

iris = px.data.iris()
px.strip(iris, 'species', 'sepal_length', 'sepal_width', stripmode='overlay')
```

![image](https://user-images.githubusercontent.com/4560057/69958881-07055b00-1506-11ea-8f14-6bec363c522e.png)

The resulting plot is not meaningful since it colors a continuous variable with a discrete colormap.

So far, I have not found a prettier workaround that this (which is imperfect since it uses the legend instead of a colorbar):

```python
import matplotlib as mpl

sw = iris['sepal_width'].sort_values()
sw_01 = (sw - sw.min()) / (sw.max() - sw.min())
sw_colors = {n: mpl.colors.rgb2hex(c) for n, c in zip(sw, mpl.cm.viridis(sw_01))}
px.strip(iris, 'species', 'sepal_length', 'sepal_width',
         stripmode='overlay', category_orders={'sepal_width': sw.to_list()[::-1]},
         color_discrete_map=sw_colors)
```

![image](https://user-images.githubusercontent.com/4560057/69958970-4cc22380-1506-11ea-9654-a30e77e0540b.png)

**Comment by nicolaskruchten** [comment]
Hi Joel!

What you're trying to do makes sense in principle, although unfortunately the changes required to implement it all sit at the Javascript level, in plotly.js. `strip` is built on top of `box` at the moment, so basically we would need to add the ability for the *points* in `box`/`violin` traces to be coloured differently from the boxes/violins themselves. If that were done, then we would get the `coloraxis` attribute in the `graph_objects` automatically (via code-generation) and we could then figure out how to bolt this onto PX.

Another approach would be to add `jitter` options to `scatter` traces, and we could then build something like `strip` on top of `scatter` instead of on top of `box`. Either way that would require changes in the JS layer again.

**Comment by nicolaskruchten** [comment]
PS: your workaround is pretty good actually! If you bin the continuous variable in Python and then color by that you can get pretty close!

**Comment by joelostblom** [comment]
Thanks for the reply @nicolaskruchten ! I suspected that this would have to be added in the js layer, but I wasn't sure how to provide example code in js so I decided to open an issue here to start. I think this would be great functionality to add to plotly, should I open a separate issue in plotly.js and reference this one?

To me, the first solution you suggested above would be preferable since it is then possible to also color the points when used together with violin or box.

**Comment by YukunYangNPF** [comment]
Any follow-ups about this issue? It seems like this continuous color scale issue for box plot is not resolved.

**Comment by nicolaskruchten** [comment]
Correct, we have not implemented this feature and no one is working on it at the moment.

**Comment by agt24** [comment]
+1 on adding this functionality. 

Curious if anyone has new/different work around to accomplish dots with a continuous color scale.

**Comment by TheAustinator** [comment]
+1

**Comment by brianmcv-elvie** [comment]
Yep this would be extremely useful!

**Comment by ethantenison** [comment]
+ 3!

**Comment by wyss-alejandra** [comment]
This feature would be helpful; I'm trying to replicate a summary plot for SHAP values, and I have not been able to do it.

**Comment by ethantenison** [comment]
> This feature would be helpful; I'm trying to replicate a summary plot for SHAP values, and I have not been able to do it.

Same! It's a shame.

**Comment by dandolodavid** [comment]
Hello! The same is for me. I'm trying to create a new feature importance visualization and I will need this. Hoper there will be a fix soon.
Thanks

**Comment by jainraj** [comment]
+1

**Comment by rabyj** [comment]
Has another workaround been found? I need the points to be colored differently than the violin part. Is there any way to iterate through fig.data to modify values one by one? Either with the marker size or the marker color?

**Comment by cegutica** [comment]
Check out [this post](https://stackoverflow.com/questions/55447131/how-to-add-a-colorbar-to-an-already-existing-plotly-figure) to complement this workaround.

**Comment by guillaume-vignal** [comment]
+1

**Comment by GillianGrayson** [comment]
+1

**Comment by SimonKnorke** [comment]
+1

**Comment by awaismirza92** [comment]
+1

**Comment by jleemartin** [comment]
+1

**Comment by jlumpe** [comment]
+1
